apps/gitness/src/pages-v2/pipelines/repo-pipelines-list.tsx

```diff
@@ -0,0 +1,108 @@
+import { Link } from 'react-router-dom'
+
+import { parseAsInteger, useQueryState } from 'nuqs'
+
+import { Button, Spacer, Text } from '@harnessio/canary'
+import { TypesPipeline, useListPipelinesQuery } from '@harnessio/code-service-client'
+import { SkeletonList } from '@harnessio/ui/components'
+import { PipelineList } from '@harnessio/ui/views'
+import { Filter, NoData, NoSearchResults, PaginationComponent, SandboxLayout } from '@harnessio/views'
+
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import { useDebouncedQueryState } from '../../hooks/useDebouncedQueryState'
+import { PageResponseHeader } from '../../types'
+import { getExecutionStatus, getMeterState } from '../../utils/execution-utils'
+
+export default function RepoPipelinesListPage() {
+  const repoRef = useGetRepoRef()
+  const [query, setQuery] = useDebouncedQueryState('query')
+  const [page, setPage] = useQueryState('page', parseAsInteger.withDefault(1))
+
+  const { data: { body: pipelines, headers } = {}, isFetching } = useListPipelinesQuery(
+    {
+      repo_ref: repoRef,
+      queryParams: { page, query, latest: true }
+    },
+    { enabled: !!repoRef }
+  )
+
+  const totalPages = parseInt(headers?.get(PageResponseHeader.xTotalPages) || '')
+
+  const LinkComponent = ({ to, children }: { to: string; children: React.ReactNode }) => <Link to={to}>{children}</Link>
+
+  const renderListContent = () => {
```

Comment:
this might run into flickering issue
https://harness.slack.com/archives/C07QA52CWKH/p1733729112788079?thread_ts=1733525285.482889&cid=C07QA52CWKH